src/sonic-bgpcfgd/tests/test_allow_list.py

```diff
@@ -0,0 +1,296 @@
+            'no route-map ALLOW_LIST_DEPLOYMENT_ID_5_V6 permit 10',
+            'no ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4',
+            'no ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6',
+            'no bgp community-list standard COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+        ]
+    cfg_mgr = MagicMock()
+    cfg_mgr.update.return_value = None
+    cfg_mgr.push_list = push_list
+    cfg_mgr.get_text.return_value = [
+        'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4 seq 10 deny 0.0.0.0/0 le 17',
+        'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4 seq 20 permit 10.20.30.0/24 ge 25',
+        'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4 seq 30 permit 30.50.0.0/16 ge 17',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 10 deny 0::/0 le 59',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 20 deny 0::/0 ge 65',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 30 permit fc00:20::/64 ge 65',
+        'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 40 permit fc00:30::/64 ge 65',
+        'bgp community-list standard COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020 permit 1010:2020',
+        'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V4 permit 10',
+        ' match ip address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4',
+        ' match community COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+        'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V6 permit 10',
+        ' match ipv6 address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6',
+        ' match community COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+        ""
+    ]
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr':   cfg_mgr,
+        'tf':        TemplateFabric(),
+        'constants': global_constants,
+    }
+    mgr = BGPAllowListMgr(common_objs, "CONFIG_DB", "BGP_ALLOWED_PREFIXES")
+    mgr.del_handler("DEPLOYMENT_ID|5|1010:2020")
+    assert test_del_handler_with_community.push_list_called, "cfg_mgr.push_list wasn't called"
+
+#@pytest.mark.skip()
+@patch('app.allow_list.run_command', side_effect=new_run_command)
+def test_del_handler_no_community(mock_run_command):
+    test_del_handler_no_community.push_list_called = False
+    def push_list(args):
+        test_del_handler_no_community.push_list_called = True
+        assert args == [
+            'no route-map ALLOW_LIST_DEPLOYMENT_ID_5_V4 permit 30000',
+            'no route-map ALLOW_LIST_DEPLOYMENT_ID_5_V6 permit 30000',
+            'no ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V4',
+            'no ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_empty_V6',
+        ]
+    cfg_mgr = MagicMock()
```

Comment:
i feel the test setup can be commonly shared and using fixture so that you can simplified the test case a little bit.

Reply:
Done.